src/common/AndroidUtils.ts

```diff
@@ -333,61 +321,66 @@ export class AndroidSDKUtils {
         device: string,
         abi: string
     ): Promise<boolean> {
-        const createAvdCommand = `${AndroidSDKUtils.getAvdManagerCommand()} create avd -n ${emulatorName} --force -k ${AndroidSDKUtils.systemImagePath(
+        // Just like Android Studio AVD Manager GUI interface, replace blank spaces with _ so that the ID of this AVD
+        // doesn't have blanks (since that's not allowed). AVD Manager will automatially replace _ back with blank
+        // to generate user friendly display names.
+        const resolvedName = emulatorName.replace(/ /gi, '_');
+
+        const createAvdCommand = `${AndroidSDKUtils.getAvdManagerCommand()} create avd -n ${resolvedName} --force -k ${AndroidSDKUtils.systemImagePath(
             platformAPI,
             emulatorImage,
             abi
         )} --device ${device} --abi ${abi}`;
-        return new Promise((resolve, reject) => {
```

Comment:
I'm not sure why we were previously using `spawnChild`.... but it was problematic. If the command for the spawned process failed for some reason, the callback was not being invoked and we would proceed as if it had succeeded. So even if the create command had failed, we would report it as succeeded.

I switched to using (synchronous) `executeCommand` and it seems to be working fine. I verified it on latest version of Android Studio as well as an older version from 2018 and all is well.